src/utils/hooks.js

```diff
@@ -0,0 +1,11 @@
+import { useState } from 'preact/hooks';
+
+/**
+ * Toggle custom hook
+ * @param {Boolean} initialState
+ * @returns {Array} flag | toggle
+ */
+export const useToggle = initialState => {
+  const [state, setState] = useState(initialState);
+  return [state, () => setState(!state)];
```

Comment:
(nit) please directly returning new assignments/literals. Instead: please assign the return value on the previous line and then return that `const`. While it may seem unnecessary (and some editors will complain that it's redundant) it helps quite a bit when debugging code.

```js
const result = [state, () => setState(!state)];
return result;
```